src/systems/filecoin_vm/runtime/runtime.go

```diff
@@ -1,81 +1,424 @@
+			rt._sendInternal(item.to, item.input, item.ignoreErrors)
+		}
+		rt._messageQueue_ = []MessageQueueItem{}
+	})
+	rt._running_ = false
+
+	if catchRetCode.IsSuccess() {
+		if rt._output() == nil {
+			panic("Internal error: return call recorded but no output object returned")
+		}
+		ret = rt._output()
+	} else {
+		ret = msg.InvocOutput_Make(catchRetCode, nil)
+	}
+
+	// TODO: Update gasUsed
+	TODO()
+
+	return
+}
+
+func (rtOuter *Runtime_I) _sendInternal(to addr.Address, input InvocInput, ignoreErrors bool) msg.MessageReceipt {
+	rtOuter._checkRunning()
+	rtOuter._checkStateLock(false)
+
+	toActor := rtOuter._globalStatePending().GetActor(to).State()
+
+	toActorCode, err := loadActorCode(toActor.CodeCID())
+	if err != nil {
+		rtOuter._throwError(exitcode.SystemError(exitcode.ActorCodeNotFound))
+	}
+
+	var toActorMethodGasBound msg.GasAmount
+	TODO() // TODO: obtain from actor registry
+	rtOuter._deductGasRemaining(toActorMethodGasBound)
+	// TODO: gasUsed may be larger than toActorMethodGasBound if toActor itself makes sub-calls.
+	// To prevent this, we would need to calculate the gas bounds recursively.
+
+	rtOuter._transferFunds(rtOuter._actorAddress(), to, input.Value())
+	rtOuter._incrementCallSeqNum(to)
+
+	rtInner := Runtime_Make(
+		rtOuter._globalStatePending(),
+		to,
+		input.Value(),
+		rtOuter._gasRemaining(),
+	)
+
+	invocOutput, gasUsed := _invokeMethodInternal(
+		rtInner.Impl(),
+		toActorCode,
+		input.Method(),
+		input.Params(),
+	)
+
+	rtOuter._refundGasRemaining(toActorMethodGasBound)
+	rtOuter._deductGasRemaining(gasUsed)
+
+	if !ignoreErrors && invocOutput.ExitCode().IsError() {
+		rtOuter._throwError(exitcode.SystemError(exitcode.MethodSubcallError))
+	}
+
+	if invocOutput.ExitCode().AllowsStateUpdate() {
+		rtOuter._globalStatePending_ = rtInner._globalStatePending()
+	}
+
+	return msg.MessageReceipt_Make(invocOutput, gasUsed)
+}
+
+func (rt *Runtime_I) Send(to addr.Address, input InvocInput) msg.MessageReceipt {
+	return rt._sendInternal(to, input, false)
+}
+
+func (rt *Runtime_I) SendAllowingErrors(to addr.Address, input InvocInput) msg.MessageReceipt {
+	return rt._sendInternal(to, input, true)
+}
+
+type MessageQueue []MessageQueueItem
+type MessageQueueItem struct {
+	to           addr.Address
+	input        InvocInput
+	ignoreErrors bool
+}
+
+func MessageQueueItem_Make(to addr.Address, input InvocInput, ignoreErrors bool) MessageQueueItem {
+	return MessageQueueItem{
+		to:           to,
+		input:        input,
+		ignoreErrors: ignoreErrors,
+	}
+}
+
+func (rt *Runtime_I) _deferredSendInternal(to addr.Address, input InvocInput, ignoreErrors bool) {
+	rt._checkRunning()
+	rt._checkStateLock(false)
+	rt._messageQueue_ = append(rt._messageQueue(), MessageQueueItem_Make(to, input, ignoreErrors))
+}
+
+func (rt *Runtime_I) DeferredSend(to addr.Address, input InvocInput) Runtime_DeferredSend_FunRet {
+	rt._deferredSendInternal(to, input, false)
+	return &Runtime_DeferredSend_FunRet_I{}
```

Comment:
what actor calls need `DeferredSend`?  do we strictly need this now? (i dont think we do)

if not, we should keep this optimization out-- it's a source of complexity (requires the message queue, which itself complicates the runtime). we may find we need it back, and can bring it back then.

Reply:
(removed deferred send for now as per discussion)